File: 6.py
#Sum square difference
#
#Problem 6
#
#The sum of the squares of the first ten natural numbers is, 1^2 + 2^2 + ... + 10^2 = 385
#
#The square of the sum of the first ten natural numbers is, (1 + 2 + ... + 10)^2 = 55^2 = 3025
#
#Hence the difference between the sum of the squares of the first ten natural numbers and the square of the sum is
#
#
#Find the difference between the sum of the squares of the first one hundred natural numbers and the square of the sum.

def sumofsquares(n):
    sum = 0
    for i in range(1, n+1):
        sum += i**2
    return sum

def squareofsum(n):
    sum = 0
    for i in range(1, n+1):
        sum += i
    return sum**2

print(squareofsum(100)-sumofsquares(100))

File: 6.c
#include <stdio.h>
#include <math.h>

int sumofsquares(int n){
    int sum = 0;
    for (int i = 0; i < n+1; i++)
    {
        sum += i * i; 
    }
    return sum;
}

int squareofsum(int n){
    int sum = 0;
    for (int i = 0; i < n+1; i++)
    {
        sum += i;
    }
    sum = sum * sum;
    return sum;


}

int main()
{
    int j = squareofsum(100)-sumofsquares(100);
    printf("%d",j);
    return 0;
}
